flang/lib/Optimizer/Transforms/FunctionAttr.cpp

```diff
@@ -45,14 +50,32 @@ void FunctionAttrPass::runOnOperation() {
     func->setAttr("frame_pointer", mlir::LLVM::FramePointerKindAttr::get(
                                        context, framePointerKind));
 
+  if (noInfsFPMath)
+    func->setAttr("no_infs_fp_math", mlir::BoolAttr::get(context, true));
```

Comment:
I am not a fan of using literal constants while they have to be kept in sync between multiple components.  Is it reasonable to make them named attributes in LLVM dialect and provide the name getters for them in `LLVMDialect.td`?

Reply:
Adding a function to get the attribute name from LLVMDialect.td is also not completely save since the attribute name on LLVMFuncOp is generated based on the ODS operation definition. That means things could still get out of sync.

Could it potentially make sense to run this pass after lowering to LLVMFuncOp? It would have setters for the attributes. Alternatively, LLVMFuncOp has getters to obtain the attribute names.

Reply:
I'm happy to go with whatever is preferred here, but I agree with @gysit that it's not clear adding such getters would be a big win. We're emitting the attribute once in our code base, and if it was typoed we'd see it fail to show up (due to being unrecognised) in the test case I added.

Reply:
My main concern is that people may change the naming of the MLIR attributes in LLVM dialect, and do not run any Flang testing (I think it is a usual practice to limit the testing to the components that your changes affect).  Flang LIT testing will break after the commit, while it is preferable not to have the breakage in the first place.  So I would prefer any solution that breaks the names dependency between Flang and LLVM dialect. Using the getters from LLVMFuncOp sounds acceptable. Moving this pass after the `FIRToLLVMLowering` pass seems to be an option as well.

Reply:
@vzakhari Though it's probably common to not build-test flang as well in that kind of scenario. I'm not sure I see any appropriate static getters in LLVMFuncOp that would get this information, and I don't believe I can get an instance of an LLVMFuncOp at this stage (though could be wrong - I'm new to MLIR and Flang). Could you please point to what you were imagining I could use?

Reply:
After #79812 we should get static methods in `LLVMFuncOp` class that return the required attribute names, e.g. `getUnsafeFpMathAttrName`, `getNoNansFpMathAttrName`, etc.  You may look it up in `tools/mlir/include/mlir/Dialect/LLVMIR/LLVMOps.h.inc` in your llvm build location.